File: FrontEnd/src/app/models/solver-output.ts
import { CalculatorOutput } from "./calculator-output";
import { Profile } from "./profile.model";

export class SolverOutput {

    public profile: Profile;
    public error: string;
    public calculatorOutput: CalculatorOutput
    public galLost: number;

    constructor(profile: Profile, error: string, calculatorOutput: CalculatorOutput, galLost: number) {
        this.profile= profile;
        this.error = error;
        this.calculatorOutput = calculatorOutput;
        this.galLost = galLost;
    }
}

File: java/src/main/java/com/opl/aircraftperformancecalculator/models/SolverOutput.java
package com.opl.aircraftperformancecalculator.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Output of solver to front end.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SolverOutput {

    private Profile profile;
    private String error; // Returned if plane is too heavy for runway.
    private CalculatorOutput calculatorOutput;
    double galLost; // Gallons needed to be dropped to land
}
